metadata-ingestion/src/datahub/ingestion/sink/file.py

```diff
@@ -27,6 +28,14 @@ def _to_obj_for_file(
 ) -> dict:
     if isinstance(obj, MetadataChangeProposalWrapper):
         return obj.to_obj(simplified_structure=simplified_structure)
+    elif isinstance(obj, MetadataChangeProposal) and simplified_structure:
+        serialized = obj.to_obj()
+        if serialized.get("aspect") and serialized["aspect"].get("contentType") in [
+            JSON_CONTENT_TYPE,
+            JSON_PATCH_CONTENT_TYPE,
+        ]:
```

Comment:
can we also add a test to tests/unit/serde/test_serde.py

we already have some stuff for patches in there

Reply:
we'll specifically need a test for reading the old serialized format and producing the new unpacked format, if we don't have that already